packages/calcite-components/src/components/action-menu/action-menu.tsx

```diff
@@ -401,7 +401,21 @@ export class ActionMenu implements LoadableComponent {
       .assignedElements({
         flatten: true,
       })
-      .filter((el) => el?.matches("calcite-action")) as HTMLCalciteActionElement[];
+      .reduce((previousValue: HTMLCalciteActionElement[], currentValue) => {
+        if (currentValue?.matches("calcite-action")) {
+          return [...previousValue, currentValue];
```

Comment:
Could we we do `return previousValue.push(currentValue)` for better performance? I don't think we need to do a shallow clone here by spreading.

Reply:
Yep!